File: JavaLibrary/src/main/java/com/flat/simplex/parser/logic/LineParser.java
package com.flat.simplex.parser.logic;

import com.flat.simplex.lexer.Token;
import com.flat.simplex.parser.logic.line.LineValue;

public class LineParser {
    
    private final Block parent;
    private final Token token;
    private final Token end;

    public LineParser(Block parent, Token token, Token end) {
        this.parent = parent;
        this.token = token;
        this.end = end;
    }

    public LineValue parse() {
        LineReader reader = new LineReader(parent);
        LineBinder binder = new LineBinder(parent, token);
        return binder.bind(reader.read(token, end));
    }
}

File: SimplexApp/simplex/lang/parser/logic/LineParser.js
var LineReader, LineBinder;

class LineParser {
    constructor(parent, token, end) {
        this.parent = parent;
        this.token = token;
        this.end = end;
    }
    parse() {
        let reader = new LineReader(this.parent);
        let binder = new LineBinder(this.parent, this.token);
        return binder.bind(reader.read(this.token, this.end));
    }
}

module.exports = LineParser;
LineReader = require("simplex/lang/parser/logic/LineReader.js");
LineBinder = require("simplex/lang/parser/logic/LineBinder.js");